src/GraphQLParser/Extensions/ASTNodeExtensions.cs

```diff
@@ -0,0 +1,37 @@
+using GraphQLParser.AST;
+using GraphQLParser.Visitors;
+
+namespace GraphQLParser
+{
+    /// <summary>
+    /// Extension methods for <see cref="ASTNode"/>.
+    /// </summary>
+    public static class ASTNodeExtensions
+    {
+        /// <summary>
+        /// Counts the number of all nested nodes of the specified node along with that node itself.
+        /// </summary>
+        /// <param name="node">Parent node.</param>
+        /// <returns>The number of nodes.</returns>
+        public static int Count(this ASTNode node)
+        {
+            var visitor = new CountVisitor<DefaultCountContext>();
+            var context = new DefaultCountContext(_ => true);
+            visitor.Visit(node, context).GetAwaiter().GetResult(); // it's safe since method is actually sync
+            return context.Count;
+        }
+
+        /// <summary>
+        /// Gets the maximum depth for this AST node.
```

Comment:
We should probably add notes that say that this is the depth of the object hierarchy, etc.

Reply:
/// Gets the maximum depth of all nested nodes' chains for this AST node.